sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/SensorContextUtilsTest.java

```diff
@@ -157,6 +159,57 @@ public void hasFilesOfLanguage_whenOnlyOtherLanguageExists_returnsFalse() {
     assertThat(hasFilesOfLanguage(fs, LANG_ONE)).isFalse();
   }
 
+  @Test
+  public void toTextRange_whenTokensStartAtEOL_doesNotFilterOut() {
+    var inputFile = new TestInputFileBuilder("mod", "source.cs")
+      .setLanguage("cs")
+      .setType(Type.MAIN)
+      .setContents(
+        "var rawString = $$\"\"\"\n" +
+        "    {\n" +
+        "        return {{BuildSwitchExpression(ref enumData)}}\n" +
+        "    }\n" +
+        "\n" +
+        "}\n" +
+        "\"\"\";")
```

Comment:
Since the `toTextRange` method works with text ranges and doesn't know about tokens we could have simpler code snippets in the test cases.

Reply:
UTs simplified with non-C#-valid examples that are more readable.